### PR TITLE
Introduce set of excluded tests for Java 10

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

For more information see https://github.com/gradle/gradle-private/issues/1066.

Jetty is not part of Gradle core anymore and therefore should run fine. I removed those tests from the exclusion list.

### CODE REVIEW

**Comment by jjohannes** [comment]
You should [sign your commits](https://help.github.com/articles/signing-commits-using-gpg/) @bmuschko to get rid of the :x: **DCO** above. :)

**Comment by bmuschko** [comment]
> You should sign your commits @bmuschko to get rid of the ❌ DCO above. :)

Hhhhm, somehow my IDE seems to have lost the ability. Will look into it.

> Although I would prefer if we keep one list (or map) of excluded tests - like a mapping of test names to a set of Java versions. Instead of duplicating. Because I wonder where we are heading with the next Java releases...

Let me see if I can make this happen right now.

File: gradle/java9plus.gradle [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    def testJavaInstallation = new DefaultJavaInstallation()
+    gradle.services.get(JavaInstallationProbe)
+        .checkJdk(new File(testJavaHome))
+        .configure(testJavaInstallation)
+    def currentJavaVersion = testJavaInstallation.javaVersion
+
+    if (isSupportedJavaVersion(currentJavaVersion)) {
+
+        tasks.withType(org.gradle.testing.IntegrationTest) {
+            ext.excludedTests = [
```
Comment (eskatos) [inline gradle/java9plus.gradle]: :x: This should be a simple `def` in order not to hold the map

**Comment by bmuschko** [comment]
@jjohannes @eskatos I made additional changes. Can you have another look?

File: gradle/java9plus.gradle (hunk 1 above)
Comment (bmuschko) [reply gradle/java9plus.gradle]: done

**Pull request merged**